src/GraphQL/Internal/Validation.hs

```diff
@@ -386,12 +472,77 @@ resolveFragmentDefinitions allFragments =
           modify (Set.insert name)
           FragmentSpread name directives <$> resolveFragment' definition
 
+-- * Variables
+
+data VariableDefinition = VariableDefinition Variable AST.Type (Maybe Value) deriving (Eq, Ord, Show)
```

Comment:
Ah I see now, thanks. Can you copy-paste your first paragraph as a comment? Maybe it's worth making this a record so we can name the fields even if we don't use the accessors?